radicle/src/ssh.rs

```diff
@@ -184,6 +241,146 @@ impl radicle_ssh::key::Private for SecretKey {
+    Base64Encoding(#[from] base64::DecodeError),
+    #[error("wrong preamble")]
+    MagicPreamble([u8; 6]),
+    #[error("missing armored footer")]
+    MissingFooter,
+    #[error("missing armored header")]
+    MissingHeader,
+    #[error(transparent)]
+    Encoding(#[from] encoding::Error),
+    #[error("public key encoding")]
+    PublicKeyEncoding,
+    #[error(transparent)]
+    PublicKeyError(#[from] PublicKeyError),
+    #[error("signature encoding")]
+    SignatureEncoding,
+    #[error(transparent)]
+    SignatureError(#[from] SignatureError),
+    #[error("unsupported version '{0}'")]
+    UnsupportedVersion(u32),
+}
+
+/// An SSH signature's decoded format.
+#[derive(Clone, Debug)]
+pub struct ExtendedSignature {
+    sig_version: u32,
+    publickey: crypto::PublicKey,
+    // unambigious interpretation domain to prevent cross-protocol attacks.
+    // Might be similar to 'git'.  Must be non-empty.
+    namespace: Vec<u8>,
+    // Should ignore
+    reserved: Vec<u8>,
+    // Hash used for signature
+    //  - sha256 or sha512
+    hash_algorithm: Vec<u8>,
+    signature: crypto::Signature,
+}
+
+impl Encodable for ExtendedSignature {
+    type Error = ExtendedSignatureError;
+
+    fn read_ssh(r: &mut encoding::Cursor) -> Result<Self, Self::Error> {
+        let sig_version = r.read_u32()?;
+        if sig_version > 1 {
+            return Err(ExtendedSignatureError::UnsupportedVersion(sig_version));
+        }
+
+        Ok(ExtendedSignature {
+            sig_version,
+            publickey: PublicKey::read_ssh(r)
+                .map_err(|_| ExtendedSignatureError::PublicKeyEncoding)?,
+            namespace: r.read_string()?.into(),
+            reserved: r.read_string()?.into(),
+            hash_algorithm: r.read_string()?.into(),
+            signature: crypto::Signature::read_ssh(r)
+                .map_err(|_| ExtendedSignatureError::PublicKeyEncoding)?,
+        })
+    }
+
+    fn write_ssh<E: Encoding>(&self, buf: &mut E) {
+        buf.extend_u32(self.sig_version);
+        let _ = &self.publickey.write_ssh(buf);
+        buf.extend_ssh_string(&self.namespace);
+        buf.extend_ssh_string(&self.reserved);
+        buf.extend_ssh_string(&self.hash_algorithm);
+        let _ = &self.signature.write_ssh(buf);
+    }
+}
+
+impl ExtendedSignature {
+    const ARMORED_HEADER: &[u8] = b"-----BEGIN SSH SIGNATURE-----";
+    const ARMORED_FOOTER: &[u8] = b"-----END SSH SIGNATURE-----";
+    const ARMORED_WIDTH: usize = 70;
+    const MAGIC_PREAMBLE: &[u8] = b"SSHSIG";
+
+    pub fn from_armored(s: &[u8]) -> Result<Self, ExtendedSignatureError> {
+        let s = s
+            .strip_prefix(Self::ARMORED_HEADER)
+            .ok_or(ExtendedSignatureError::MissingHeader)?;
+        let s = s
+            .strip_suffix(Self::ARMORED_FOOTER)
+            .ok_or(ExtendedSignatureError::MissingFooter)?;
+        let s: Vec<u8> = s
+            .iter()
+            .filter_map(|b| if b != &b'\n' { Some(*b) } else { None })
```

Comment:
Seems like you're just returning `b` in your filter_map, so you may as well use `filter(|b| *b != b'\n')`